third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -2730,19 +2826,396 @@ impl Frame {
+                            (Ok(LoadedFunction {
+                                owner: LoadedFunctionOwner::Module(module.clone()),
+                                ty_args: verified_ty_args,
+                                function: function.clone(),
+                            }))
+                        },
+                        FunctionHandle::Remote { module, name } => self
+                            .build_loaded_function_from_name_and_ty_args(
+                                module_storage,
+                                module,
+                                name,
+                                verified_ty_args,
+                            ),
+                    }
+                },
+                LoadedFunctionOwner::Script(script) => {
+                    let handle = script.$get_function_handle(idx.0);
+                    match handle {
+                        FunctionHandle::Local(_) => Err(PartialVMError::new(
+                            StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                        )
+                        .with_message("Scripts never have local functions".to_string())),
+                        FunctionHandle::Remote { module, name } => self
+                            .build_loaded_function_from_name_and_ty_args(
+                                module_storage,
+                                module,
+                                name,
+                                verified_ty_args,
+                            ),
+                    }
+                },
+            }
+        }
+    };
+}
+
+impl Frame {
+    build_loaded_function!(
+        build_loaded_function_from_handle_and_ty_args,
+        FunctionHandleIndex,
+        function_at
+    );
+
+    build_loaded_function!(
+        build_loaded_function_from_instantiation_and_ty_args,
+        FunctionInstantiationIndex,
+        function_instantiation_handle_at
+    );
+
+    pub(crate) fn build_loaded_function_from_name_and_ty_args(
+        &self,
+        module_storage: &impl ModuleStorage,
+        module_id: &ModuleId,
+        function_name: &IdentStr,
+        verified_ty_args: Vec<Type>,
+    ) -> PartialVMResult<LoadedFunction> {
+        let (module, function) = module_storage
+            .fetch_function_definition(module_id.address(), module_id.name(), function_name)
+            .map_err(|_| {
+                // Note: legacy loader implementation used this error, so we need to remap.
+                PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE).with_message(format!(
+                    "Module or function do not exist for {}::{}::{}",
+                    module_id.address(),
+                    module_id.name(),
+                    function_name
+                ))
+            })?;
+        Ok(LoadedFunction {
+            owner: LoadedFunctionOwner::Module(module),
+            ty_args: verified_ty_args,
+            function,
+        })
+    }
```

Comment:
Just to confirm, this is basically copy-paste from `Resolver`, right?

Reply:
Yes!